### PR TITLE
feat(accounts): Add redirect handling to /accounts/social SignupView

### PR DESCRIPTION
**Pull request by denisemauldin**
# Submitting Pull Requests

## General

 - [x] Make sure you use [semantic commit messages](https://seesparkbox.com/foundry/semantic_commit_messages).
       Examples: `"fix(google): Fixed foobar bug"`, `"feat(accounts): Added foobar feature"`.
 - [x] All Python code must formatted using Black, and clean from pep8 and isort issues.
 - [ ] JavaScript code should adhere to [StandardJS](https://standardjs.com).
 - [ ] If your changes are significant, please update `ChangeLog.rst`.
 - [ ] If your change is substantial, feel free to add yourself to `AUTHORS`.

 ## Provider Specifics

 In case you add a new provider:

- [ ] Make sure unit tests are available.
- [ ] Add an entry of your provider in `test_settings.py::INSTALLED_APPS` and `docs/installation.rst::INSTALLED_APPS`.
- [ ] Add documentation to `docs/providers.rst`.
- [ ] Add an entry to the list of supported providers over at `docs/overview.rst`.

### CODE REVIEW

**Comment by denisemauldin** [comment]
@pennersr Not sure if tests are necessary, but happy to write some based off suggestions.

**Comment by pennersr** [comment]
When the form is valid, the social signup view ends with a call to: `helpers.complete_social_signup()`.  Inside that method, the next URL stashed in the `sociallogin` is already used (`sociallogin.get_redirect_url(request)`), which is identical to the `next = sociallogin.state.get(REDIRECT_FIELD_NAME, None)` call that you are adding. So redirects should already be properly be handled. if not, then we definitely could use a test case to expose the problem you are trying to address here.

File: allauth/socialaccount/helpers.py [hunk 1]
```diff
@@ -25,6 +26,12 @@ def _process_signup(request, sociallogin):
     if not auto_signup:
         request.session["socialaccount_sociallogin"] = sociallogin.serialize()
         url = reverse("socialaccount_signup")
+        next = sociallogin.state.get(REDIRECT_FIELD_NAME, None)
```
Comment (pennersr) [inline allauth/socialaccount/helpers.py:29]: Why not call `sociallogin.get_redirect_url(request)`?

File: allauth/socialaccount/helpers.py [hunk 2]
```diff
@@ -25,6 +26,12 @@ def _process_signup(request, sociallogin):
...
+        if next:
+            url = "{}?{}={}".format(
+                url, REDIRECT_FIELD_NAME, next
```
Comment (pennersr) [inline allauth/socialaccount/helpers.py:32]: This would need `urlencode()`'ing

**Review by pennersr — CHANGES_REQUESTED** [review]

**Pull request closed without merging**